kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -1118,10 +1120,24 @@ public Optional<List<String>> getClusteringPartitionFieldNames() {
   }
 
   public Optional<List<String>> getConvertMapFieldsToString() {
-    return Optional
-            .ofNullable(getList(CONVERT_MAP_FIELDS_TO_STRING_CONFIG))
-            // With Java 11 there's Predicate::not, but for now we have to just manually invert the isEmpty check
-            .filter(l -> !l.isEmpty());
+    return Optional.ofNullable(parseConvertMapFieldsToStringConfig(getString(CONVERT_MAP_FIELDS_TO_STRING_CONFIG)));
+  }
+
+  private List<String> parseConvertMapFieldsToStringConfig(String ConvertMapFieldsToStringConfig) {
+    if (ConvertMapFieldsToStringConfig.isEmpty()) {
+      return null;
+    }
+    List<String> ConvertMapFieldsToString = new ArrayList<>();
+
+    for (String str : ConvertMapFieldsToStringConfig.split(";")) {
+      String[] topicFieldsString = str.split(":");
+      String topic = topicFieldsString[0].trim();
+      String fields = topicFieldsString[1].trim();
+      for (String field : fields.split(",")) {
+        ConvertMapFieldsToString.add(topic + ":" + field.trim());
+      }
+    }
+    return ConvertMapFieldsToString.isEmpty() ? null : ConvertMapFieldsToString;
```

Comment:
try to stream and util method as below:

return Arrays.stream(ConvertMapFieldsToStringConfig.split(";")).flatMap(ele -> {
      String[] topic2Fields = ele.split(":");
      String topic = topic2Fields[0].trim();
      return Arrays.stream(topic2Fields[1].trim().split(",")).map(field -> topic + "." + field.trim());
    }).collect(Collectors.toList());